pgjdbc/src/test/java/org/postgresql/jdbc/ResourceLockTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.LockSupport;
+
+class ResourceLockTest {
+
+  @Test
+  void testObtainClose() {
+    final ResourceLock lock = new ResourceLock();
+
+    assertFalse(lock.isLocked());
+    assertFalse(lock.isHeldByCurrentThread());
+
+    try (ResourceLock ignore = lock.obtain()) {
+      assertTrue(lock.isLocked());
+      assertTrue(lock.isHeldByCurrentThread());
+    }
+
+    assertFalse(lock.isLocked());
+    assertFalse(lock.isHeldByCurrentThread());
```

Comment:
@rbygrave , my 2c here: I would love to forbid `assertTrue(boolean)` and `assertFalse(boolean)` since they produce hard to reason error messages in the CI build logs.
Of course, currently, the tests work. However, if somebody else refactors the code later, they might get an obscure failure like `expected true got false at line 38`. The worst thing is that the line number might change if the test code is modified.

It would be way easier to maintain the code if the assertions included explanation messages or if the assertions were located in their own test methods with verbose names. The stacktrace includes test method name, so having assertTrue(boolean) is not that bad if method name explains everything.

Here's a sample of what I mean:

```suggestion
    assertFalse(lock.isLocked(), "lock.isLocked(). The newly created resource lock should be unlocked");
    assertFalse(lock.isHeldByCurrentThread(), "lock.isLocked(). The newly created resource lock should be unlocked");

    try (ResourceLock ignore = lock.obtain()) {
      assertTrue(lock.isLocked(), "lock.isLocked(). We are in try ( ... lock.obtain()) section, so lock should be locked");
      assertTrue(lock.isHeldByCurrentThread(), "lock.isHeldByCurrentThread(). We are in try ( ... lock.obtain()) section, so lock should be locked");
    }

    assertFalse(lock.isLocked(), "lock.isLocked(). The lock should be released after we leave try (...lock.obtain()) section");
    assertFalse(lock.isHeldByCurrentThread(), "lock.isHeldByCurrentThread(). The lock should be released after we leave try (...lock.obtain()) section");
```

Reply:
Yes, I get your point and will have a look ...

Reply:
I have made changes to add good comments.  I'll mark this as resolved.